### PR TITLE
feature #662211

### PR DESCRIPTION
**Pull request by reeshah**
Please review and merge

### CODE REVIEW

File: BinarySearch.java [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+  public static void main(String args[])
+  {
+    int c, first, last, middle, n, search, array[];
+ 
+    Scanner in = new Scanner(System.in);
+    System.out.println("Enter number of elements");
+    n = in.nextInt(); 
+    array = new int[n];
+ 
+    System.out.println("Enter " + n + " integers");
```
Comment (bbasu) [inline BinarySearch.java]: Please user loggers instead of syso

File: BinarySearch.java [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+ 
+    first  = 0;
+    last   = n - 1;
+    middle = (first + last)/2;
+ 
+    while( first <= last )
+    {
+      if ( array[middle] < search )
+        first = middle + 1;    
+      else if ( array[middle] == search )
```
Comment (bbasu) [inline BinarySearch.java]: please nest your code properly, to make it readable

File: BinarySearch.java (hunk 1 above)
Comment (reeshah) [reply BinarySearch.java]: @bbasu removed syso

File: BinarySearch.java (hunk 2 above)
Comment (reeshah) [reply BinarySearch.java]: corrected the indentation

**Pull request merged**